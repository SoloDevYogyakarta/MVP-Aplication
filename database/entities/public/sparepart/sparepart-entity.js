"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sparepartEntity = exports.SparepartEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class SparepartEntity extends base_entity_1.BaseEntity {
}
exports.SparepartEntity = SparepartEntity;
exports.sparepartEntity = entity_1.sequelize.define('SPAREPART', {
    public_id: sequelize_1.DataTypes.STRING,
    text: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'SPAREPART',
    schema: 'public',
    timestamps: false,
});
//# sourceMappingURL=sparepart-entity.js.map