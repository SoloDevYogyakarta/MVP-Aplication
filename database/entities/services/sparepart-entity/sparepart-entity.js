"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sparepartEntity = exports.SparepartEntity = void 0;
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class SparepartEntity extends base_entity_1.BaseEntity {
}
exports.SparepartEntity = SparepartEntity;
exports.sparepartEntity = entity_1.sequelize.define('SPAREPART', {}, {
    tableName: 'SPAREPART',
    schema: 'SERVICES',
});
//# sourceMappingURL=sparepart-entity.js.map