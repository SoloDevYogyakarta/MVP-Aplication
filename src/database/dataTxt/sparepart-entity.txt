{"id":3,"public_id":"AHA1qU8mgu_mHSx1KmZhr","text":"Audeo comminor comptus. Similique demo aufero convoco caries demens. Degero ipsam ulterius copia tamquam decipio adipiscor talis pecto."}