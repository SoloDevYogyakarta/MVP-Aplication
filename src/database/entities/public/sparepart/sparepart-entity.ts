import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class SparepartEntity extends BaseEntity {
  text: string;
}

export type SparepartCreationAttribute = Optional<SparepartEntity, 'id'>;
export interface SparepartInstance
  extends Model<SparepartCreationAttribute, SparepartEntity>,
    SparepartEntity {}
export const sparepartEntity = sequelize.define<SparepartInstance>(
  'SPAREPART',
  {
    public_id: DataTypes.STRING,
    text: DataTypes.STRING,
  } as ModelAttributes<SparepartInstance>,
  {
    tableName: 'SPAREPART',
    schema: 'public',
    timestamps: false,
  },
);
