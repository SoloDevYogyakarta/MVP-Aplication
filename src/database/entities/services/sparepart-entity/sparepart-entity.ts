import { Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class SparepartEntity extends BaseEntity {}

export type SparePartCreationAttribute = Optional<SparepartEntity, 'id'>;
export interface SparepartInstance
  extends Model<SparePartCreationAttribute, SparepartEntity>,
    SparepartEntity {}
export const sparepartEntity = sequelize.define<SparepartInstance>(
  'SPAREPART',
  {} as ModelAttributes<SparepartInstance>,
  {
    tableName: 'SPAREPART',
    schema: 'SERVICES',
  },
);
