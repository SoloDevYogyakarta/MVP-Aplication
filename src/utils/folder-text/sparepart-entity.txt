{"id":11,"updatedAt":"2023-11-08T12:45:45.330Z","createdAt":"2023-11-08T12:45:45.330Z"}