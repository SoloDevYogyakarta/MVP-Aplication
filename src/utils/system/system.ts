import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
