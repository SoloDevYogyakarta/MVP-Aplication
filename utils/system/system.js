"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removepath = exports.readpath = exports.createpath = exports.joinpath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = require("path");
function joinpath(path) {
    return (0, path_1.join)(__dirname, path);
}
exports.joinpath = joinpath;
function createpath(path, data) {
    if (typeof data !== 'string') {
        data = JSON.stringify(data);
    }
    fs_1.default.writeFileSync(joinpath(path), data);
}
exports.createpath = createpath;
function readpath(path) {
    return fs_1.default.readFileSync(joinpath(path), { encoding: 'utf-8' });
}
exports.readpath = readpath;
function removepath(path) {
    fs_1.default.unlinkSync(joinpath(path));
}
exports.removepath = removepath;
//# sourceMappingURL=system.js.map