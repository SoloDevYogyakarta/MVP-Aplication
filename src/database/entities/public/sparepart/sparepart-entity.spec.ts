import { faker } from '@faker-js/faker';
import { nanoid } from 'nanoid';
import { ModelCtor } from 'sequelize';
import { createpath } from '../../../../utils/system/system';
import { sparepartEntity, SparepartInstance } from './sparepart-entity';

describe('SparepartEntity', () => {
  let public_id!: string;
  let entity: ModelCtor<SparepartInstance>;

  beforeEach(() => {
    entity = sparepartEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  it('create', async () => {
    const public_id = nanoid();
    const create = await entity.create({
      public_id,
      text: faker.lorem.paragraph(),
    });
    create.save();

    createpath('../../database/dataTxt/sparepart-entity.txt', create);
    expect(create.public_id).toEqual(public_id);
  });

  try {
  } catch (err) {
    // empty
  }

  if (public_id) {
    it('update', async () => {
      const findOne = await entity.findOne({ where: { public_id } });
      const text = faker.lorem.paragraph();
      findOne.text = text;
      findOne.save();
      expect(findOne.text).toEqual(text);
    });

    it('findOne', async () => {
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  }
});
