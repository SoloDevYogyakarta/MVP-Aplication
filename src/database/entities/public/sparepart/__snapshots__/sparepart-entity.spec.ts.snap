// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SparepartEntity render correctly 1`] = `[Function]`;
